Addtodo

SerejaJS opened the pull request.

SerejaJS committed:
add todo app

SerejaJS committed:
Update TodoApp.jsx

SergisxShoma commented on package-lock.json:
добавь в проект файл гитигнор, и помести туда этот файл, т к он компилируется каждый раз при установки зависимостей, он в гите не нужен

SerejaJS replied:
Добавил файл package-lock.json в гитигнор

SergisxShoma commented on src/App.css:
```diff
@@ -1,38 +1,11 @@
+margin-left: 25%;
+margin-top: 50px;
+border-radius: 10px;
+/* display: flex; */
+/* align-items: center; */
+/* justify-content: center; */
```
Лучше всего не пушить закомментированный код, т к это засоряет проект

SerejaJS replied:
Удалил

SergisxShoma commented on src/Components/ToDoForm/FormStyle.css:
```diff
@@ -0,0 +1,70 @@
+}
+
+.addButton {
+  /* padding: 12px 16px; */
```
аналогично

SerejaJS replied:
Удалил

SergisxShoma commented on src/Components/ToDoForm/TodoForm.jsx:
```diff
@@ -0,0 +1,49 @@
+  }
+
+  return (
+    <div>
```
нужен ли тут див, или лучше исполдьзовать реакт фрагмент?

SerejaJS replied:
Исправил, заменил div на Fragment

SergisxShoma commented on src/Components/ToDoItem/ToDoItem.css:
```diff
@@ -0,0 +1,35 @@
+  
+  .toggleButton {
+      margin-left: 10px;
+    /* width:30px ; */
```
аналогично

SerejaJS replied:
Удалил

SergisxShoma commented on src/Components/ToDoItem/TodoItem.jsx:
```diff
@@ -0,0 +1,20 @@
+export default function TodoItem({ item, deleteTask, toggleTask }) {
+  return (
+    <li
+      className={`liStyle ${item.completed ? "completed" : "red"}`}
```
эту логику лучше вынести из рендра, т е в самом компоненте определить стиль, а в рендер отдать уже переменную

SerejaJS replied:
тут не понял(

SergisxShoma commented on src/TodoApp.jsx:
```diff
@@ -0,0 +1,56 @@
+  //Изменяю статус задачи по кнопке
+  function toggleButton(id) {
+    setItems(
+      items.map(function (item) {
+        if (item.id === id) {
+          return {
+            id: item.id,
+            value: item.value,
+            completed: !item.completed,
+          };
+        }
+        return item;
+      })
```
можно написать проще

`items.map(item =>  {...item, completed: item.id === id ? !item.completed : item.completed})`

SerejaJS replied:
Заменил код.

SergisxShoma commented on src/index.js:
```diff
@@ -1,17 +1,17 @@
-);
+ReactDOM.createRoot(document.getElementById('root')).render(<TodoApp />)
+// root.render(
+//   <React.StrictMode>
+//     <TodoApp />
+//   </React.StrictMode>
+// );
 
 // If you want to start measuring performance in your app, pass a function
 // to log results (for example: reportWebVitals(console.log))
 // or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
-reportWebVitals();
+// reportWebVitals();
```
лучше убирать закомментированный код)

SerejaJS replied:
Убрал

SergisxShoma reviewed (commented):
Очень хорошо пишешь код, но есть нюансы)

SerejaJS committed:
Update App.css

Удалил из файла закомментированный код

SerejaJS committed:
Update FormStyle.css

Удалил код, который не используется

SerejaJS committed:
Update TodoForm.jsx

Заменил div на Fragment

SerejaJS committed:
Update ToDoItem.css

Удалил код, который не используется

SerejaJS committed:
Update index.js

Убрал закомментированный код

SerejaJS committed:
Update TodoApp.jsx

Сократил лишний код и записал в одну строку логику изменения статуса задачи по кнопке

SerejaJS committed:
Update .gitignore

Добавил в игнор файл

SergisxShoma reviewed (commented):
все, молодецф, можно сливать в дев
